src/coreclr/jit/morph.cpp

```diff
@@ -377,7 +377,9 @@ GenTree* Compiler::fgMorphExpandCast(GenTreeCast* tree)
 #if defined(TARGET_ARM) || defined(TARGET_AMD64)
                         return nullptr;
 #else  // TARGET_X86
-                        return fgMorphCastIntoHelper(tree, CORINFO_HELP_DBL2UINT, oper);
+                        oper = gtNewCastNode(TYP_LONG, oper, false, TYP_LONG);
```

Comment:
It would be better to leave this change to the Intel PRs too.

Reply:
The previous behaviour here was kinda a bug since `CORINFO_HELP_DBL2UINT` was bound to `Dbl2Lng`, it only worked fine cause X86 and ARM32 ABIs return the lower bits of the long same as an uint. Calling the `Dbl2UInt` helper here would be a performance regression which you said are better left to the Intel PRs.

Reply:
Yes, I agree the existing code is a hack but there is no observable bug. There is no point in cleaning it up in this PR. It requires an extra throw away work to evaluate the impact. I suspect that this change may be changing the observable behavior. We want to have the changes of observable behavior centralized in the Intel PR.

This PR is 95% good and non-controversial, and it is just these few changes that are holding it back. If you would like it to merged soon, I would recommend reverting the few controversial changes that are holding it back.

Reply:
> I suspect that this change may be changing the observable behavior.

How so? Isn't casting a long to uint the same as taking the lower 4B from it?

> these few changes

You mean the change here and the `// unused` comments?

Reply:
> Ideally, anything related to floating point to integer conversions that are being completely redone in the Intel PR.

I'd argue that cleaning things up here makes the Intel changes simpler since it reduces the amount of work they need to do there. If the Intel folks prefer it, I can remove those from here however cc @khushal1996

To avoid JIT changes in this PR, I've moved `double -> uint` casts to a managed implementation in the latest [that copies what LLVM does](https://godbolt.org/z/vKerYs1a8) since it's faster on my machine anyway. It is a bit of a throwaway work like you said but this change makes the PR simpler and I'd argue that it's safe since LLVM uses this implementation.